src/pybind/mgr/cephadm/serve.py

```diff
@@ -1249,22 +1249,28 @@ async def _create_daemon(self,
                     'Reconfiguring' if reconfig else 'Deploying',
                     daemon_spec.name(), daemon_spec.host))
 
+                cmd = [
+                    '--name', daemon_spec.name(),
+                    '--meta-json', json.dumps({
+                        'service_name': daemon_spec.service_name,
+                        'ports': daemon_spec.ports,
+                        'ip': daemon_spec.ip,
+                        'deployed_by': self.mgr.get_active_mgr_digests(),
+                        'rank': daemon_spec.rank,
+                        'rank_generation': daemon_spec.rank_generation,
+                        'extra_container_args': extra_container_args,
+                        'extra_entrypoint_args': extra_entrypoint_args
+                    }),
+                    '--config-json', '-',
+                ]
+
+                if daemon_spec.daemon_type == 'osd':
+                    osd_objectstore = getattr(self.mgr.spec_store.active_specs.get(daemon_spec.service_name), 'objectstore', 'bluestore')
```

Comment:
I think this should be handled in the other block we already have for adding additional args for osd deployments. A bit above here there is a block
```
# osd deployments needs an --osd-uuid arg
if daemon_spec.daemon_type == 'osd':
    if not osd_uuid_map:
        osd_uuid_map = self.mgr.get_osd_uuid_map()
    osd_uuid = osd_uuid_map.get(daemon_spec.daemon_id)
    if not osd_uuid:
        raise OrchestratorError('osd.%s not in osdmap' % daemon_spec.daemon_id)
    daemon_spec.extra_args.extend(['--osd-fsid', osd_uuid])
```
If there, you just add something like
```
spec = cast(DriveGroupSpec, self.mgr.spec_store[daemon_spec.service_name].spec)
objectstore = spec.objectstore
if objectstore:
    daemon_spec.extra_args.extend(['--objectstore', objectstore])
```
that should put the arg in the correct spot (note that `--osd-fsid` is another arg for the deploy command specifically and we do it the same way by adding it to `daemon_sepc.extra_args`) and it will keep all of our osd specific changes in one location.

Reply:
Thanks, have modified.

Reply:
But there is a weird thing, it can't pass 'make check'. It shows that 'Incompatible types in assignment (expression has type "DriveGroupSpec", variable has type "CustomContainerSpec")' in the Jenkins report, IMO osd daemon should not have CustomContainerSpec.

Reply:
I checked and it seems to be taking issue with us reusing the variable named `spec`. I made a sample commit that gets mypy to pass by renaming the variables to differentiate them so you'll probably need something similar https://github.com/adk3798/ceph/commit/4c55b5a38851b23a1c9f131dd4f136828cd170b8

Reply:
Sorry, I boobed, it is happy now.